internal/util-cache/src/main/scala/sbt/internal/util/AdditionalFormats.scala

```diff
@@ -0,0 +1,53 @@
+package sbt.internal.util
+
+import sbt.internal.util.Types.:+:
+
+import sjsonnew.{ Builder, deserializationError, JsonFormat, Unbuilder }
+import sjsonnew.BasicJsonProtocol, BasicJsonProtocol.asSingleton
+
+import java.io.{ ByteArrayInputStream, ByteArrayOutputStream, InputStream, OutputStream }
+
+trait HListFormat {
```

Comment:
sjson-new comes with LList (labelled heterogeneous list), so it might be make a better encoding if we project HList into that. For example we can say `String :+: Int :+: HNil` would look like `("_1", "A") :*: ("_2", 1) :*: LNil`. This would result to a flat JSON:

```
{
  "_1": "A",
  "_2": 1
}
```